src/shared/components/CTag.vue

```diff
@@ -0,0 +1,63 @@
+<template>
+  <button
+    :class="['custom-button', variantClass]"
+    type="button"
+    v-bind="$attrs"
+  >
+    <slot /><img v-if="props.icon !== 'none'" :src="props.icon" />
```

Comment:
Mesmo com essa lógica, o componente img continua sendo renderizado mesmo sem valor. Pra resolver isso, é só remover o valor padrão como comentado abaixo do icon e no v-if basta pôr icon.

```suggestion
    <slot /><img v-if="icon" :src="props.icon" />
```

Reply:
E quanto a esse componente em específico, não vamos usar slots. Pode criar uma propriedade title e por ela passaremos o texto da tag.